src/checkpoint/checkpoint_txn.c

```diff
@@ -604,15 +604,50 @@ __wt_checkpoint_progress(WT_SESSION_IMPL *session, bool closing)
     /* Time since the full database checkpoint started */
     time_diff = WT_TIMEDIFF_SEC(cur_time, conn->ckpt.ckpt_api.timer_start);
 
-    if (closing || (time_diff / WT_PROGRESS_MSG_PERIOD) > conn->ckpt.progress_msg_count) {
+    if (closing || (time_diff / WT_PROGRESS_MSG_PERIOD) > conn->ckpt.progress.msg_count) {
         __wt_verbose_info(session, WT_VERB_CHECKPOINT_PROGRESS,
           "Checkpoint %s for %" PRIu64 " seconds and wrote: %" PRIu64 " pages (%" PRIu64 " MB)",
-          closing ? "ran" : "has been running", time_diff, conn->ckpt.write_pages,
-          conn->ckpt.write_bytes / WT_MEGABYTE);
-        conn->ckpt.progress_msg_count++;
+          closing ? "ran" : "has been running", time_diff, conn->ckpt.progress.write_pages,
+          conn->ckpt.progress.write_bytes / WT_MEGABYTE);
+        conn->ckpt.progress.msg_count++;
     }
 }
 
+/*
+ * __wti_checkpoint_progress_clear --
+ *     Clear checkpoint progress data.
+ */
+void
+__wti_checkpoint_progress_clear(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
+    conn->ckpt.progress.msg_count = 0;
+    conn->ckpt.progress.write_bytes = 0;
+    conn->ckpt.progress.write_pages = 0;
+}
+
+/*
+ * __wt_checkpoint_progress_stats --
+ *     Update checkpoint progress data.
+ */
+void
+__wt_checkpoint_progress_stats(WT_SESSION_IMPL *session, uint64_t write_bytes)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
+    conn->ckpt.progress.write_bytes += write_bytes;
+    ++conn->ckpt.progress.write_pages;
+
+    /* Periodically log checkpoint progress. */
+    if (conn->ckpt.progress.write_pages % (5 * WT_THOUSAND) == 0)
+        __wti_checkpoint_progress(session, false);
```

Comment:
Is that the right move to hide this one here? Or should we decorrelate `__wt_checkpoint_progress_stats` which updates stats and this call? If we go this way, we need to introduce an API to interact with `write_pages` from the outside and then call `__wti_checkpoint_progress`.

Reply:
I suggest to add the `#public` tag to the function's comment.

Reply:
I suggest we have a dedicated PR to add those public tags, as discussed it has not been introduced here to the code.